### ISSUE TITLE
Add quarto-render action in this repo

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-actions#1]
Hi @pommevilla ! 

We have now create this repo to store action about Quarto. I have added a simple action to install quarto. 

Would you be ok to move your work for `quarto-render` into this repo ? 
I am thinking of https://github.com/pommevilla/quarto-render

We could also work on some examples for Quarto project together. What do you think ?

**Comment by pommevilla** [comment]
Yes, feel free to do so! The action currently installs a known stable version of Quarto and calls `quarto render` in the project root directory. However, it currently only supports rendering site projects. I haven't worked on it in a while, but I was last working on 

- adding an argument to render in a different directory 
- supporting other project types

You can see this work in the `feature/render-to` branch. I'll take a look at it this weekend and clean it up.

**Comment by cderv** [comment]
We are in no rush so you can finish your work. Also, you can PR directly here, and I could make you a collaborator if you want to help me with this repo as you have now some experience with Quarto and Github Action 😉 

Otherwise, I'll wait for you clean-up and port it over here later.

**Comment by cscheid** [comment]
(In case it's even remotely useful, my fork is [`@cscheid/quarto-render`](https://github.com/cscheid/quarto-render))

**Comment by pommevilla** [comment]
Finally coming back around to this. Should the `quarto-render` action install Quarto as part of its own workflow as it currently does? Or are we assuming that people will do both steps? 

If we are installing it as part of the workflow, I'll use `install-quarto` for the most recent version and a `known-stable` branch as @cscheid suggested.

Also, I read up on how to "transfer" the `quarto-render` name to this repo in the future. I was concerned that by delisting my version of `quarto-render` in the Marketplace, it would break the repos of the people currently using that action. According to [this](https://stackoverflow.com/a/65357990/6024276) and [this](https://docs.github.com/en/actions/creating-actions/publishing-actions-in-github-marketplace#removing-an-action-from-github-marketplace), though, the GHA marketplace is just an index for actions and delisting my action won't break anything in other people's repos. All this to say that when we're ready to have this be listed in the marketplace, no one's repo will break.

**Comment by cderv** [comment]
> Should the quarto-render action install Quarto as part of its own workflow as it currently does? Or are we assuming that people will do both steps?

We could separate the step, but if you need to render a document, you would need to install quarto. So it makes sense for me that the quarto render action to both step in one. The quarto render composite action could use internally the other `install-quarto` action. About Users who would want more flexibility, they could install quarto then render at command line themself. 
The only advantage I see to separate would be if you need to quarto render several times in the same workflow - in that case, separating step makes more sense unless quarto-render action knows that quarto is already installed to skip installation. 

> I'll use install-quarto for the most recent version and a known-stable branch as 

We could adapt the `install-quarto` action to add a version parameter so that you can install a specific version. And we could default the action to the last version we consider "stable")

> Also, I read up on how to "transfer" the quarto-render name to this repo in the future. I was concerned that by delisting my version of quarto-render in the Marketplace, it would break the repos of the people currently using that action. According to this and this, though, the GHA marketplace is just an index for actions and delisting my action won't break anything in other people's repos. All this to say that when we're ready to have this be listed in the marketplace, no one's repo will break.

About marketplace, I wasn't specifically planning to publish there. I didn't pay attention yours was. Glad it will still not break anything. However, do you know if storing several actions in the same repository within different folder is compatible to publishing on marketplace ? Or is there a constraint to have 1 repo = 1 action ?

**Comment by pommevilla** [comment]
> The only advantage I see to separate would be if you need to quarto render several times in the same workflow - in that case, separating step makes more sense unless quarto-render action knows that quarto is already installed to skip installation.

Good idea! Ideally, I'd like to add it as an `if` keyword to the composite action, but currently, conditionals are not supported in composite actions ([issue](https://github.com/actions/runner/issues/834)). I'll add it as a future task.

> We could adapt the install-quarto action to add a version parameter so that you can install a specific version. And we could default the action to the last version we consider "stable")

I think that's a great idea. So it sounds like there are three modes for `install-quarto`'s version variable:

- `stable` (default) - the last one we consider stable
- `dev` - the latest version
- any integer - install the user input version if it's a valid version number. If not, install the stable version.

Once it gets implemented in `install-quarto`, I'll go ahead and implement it in `quarto-render`

> However, do you know if storing several actions in the same repository within different folder is compatible to publishing on marketplace ? Or is there a constraint to have 1 repo = 1 action?

According to [the docs](https://docs.github.com/en/actions/creating-actions/publishing-actions-in-github-marketplace#about-publishing-actions) we can only have one action per repo. I guess it's not really a big deal to publish on the marketplace, though. For example, the [r-lib/actions repo](https://github.com/r-lib/actions) has a setup that looks similar to what you're going for with this repo.
